### ISSUE TITLE
Android: taking snapshot do not return actual  camera preview

### ISSUE THREAD

**Issue by DmitryAverin** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#565]
I use method `CameraPreview.takeSnapshot(function (base64PictureData) )`, because we taking not standard  size of picture and it is important feature. So method `takePicture()` does not fit for my situation. Seems on android phones and tablet plugin is making picture bigger that actual size of camera preview. I tried it on real devices and emulator. For ios I haven't notice this problem. I also attach screenShots, which show the problem
![Снимок экрана 2019-08-13 в 11 23 48](https://user-images.githubusercontent.com/52533547/62924813-edf36580-bdc1-11e9-8558-dc4c7b8dc137.png)
![Снимок экрана 2019-08-13 в 11 23 41](https://user-images.githubusercontent.com/52533547/62924814-edf36580-bdc1-11e9-81b1-d479870657a4.png)

**Comment by marcusdiy** [comment]
That's right, we have the same problem, we could have a great perfomance boost if this part given back the real size. Just for example, on android we get an 1920px image vs 460px on the preview...

**Comment by westonganger** [comment]
* PR: https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview/pull/209
* Actual Commit: https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview/commit/406e0c833cf0d2f2dd846b4840d929809260d9e6

PR #209 is what introduced this functionality. I would actually like to see the ability to take picture of the preview. If anyone wanted to take a stab at adding the old behaviour back by maybe adding an option to take pictures as preview or full size photo (current).

**Comment by marcusdiy** [comment]
@westonganger would be great. Btw documentation says:
.takeSnapshot(...) Take snapshot of the camera preview
.takePicture(...) Take the picture

**Comment by westonganger** [comment]
Sorry I forgot we added that method. `takeSnapshot` was added in PR #525

**Comment by Irahe** [comment]
Hey guys, i had the same problems for a couple of days and i have figured it out myself.Unfortunately i dont have the skills to modify this awesome plugin to behave the way we need, so i writed a simple JS code to make the work for me.It just take a offsetX, offsetY and a size(w, h) and return the new base64 image cropped.In my project i calculate the proportions from preview size so i have the same result on all device sizes and picture sizes.I hope the maintainers can eventually find a way to serve this functionality.I also hope this(amateur) code can help someone achieve something. :)



```
function takeCustomSnapshot() {
    CameraPreview.takeSnapshot({ quality: 100 }, function (data) {
        let base64Image = 'data:image/jpeg;base64,' + data;

        let offsetX = 100; //desired offset x to cut
        let offsetY = 100;//desired offset y to cut
        let resultWidth = 150;//desired x size
        let resultHeight = 150;//desired y size

        cropImage(base64Image, offsetY, offsetX, resultWidth, resultHeight, function (cropped_img_data) {
            //do something with the new cropped image

            let review_image = document.getElementById('image_review_img');//i show it on a img element...
            review_image.style.height = resultHeight + 'px';
            review_image.style.width = resultWidth + 'px';
            review_image.src = cropped_img_data;//img_data is your new croped picture...
        });
        //CameraPreview.stopCamera();
    });
}

function cropImage(base64Image, offsetY, offsetX, resultWidth, resultHeight, callback) {
    let canvas = document.createElement('canvas');
    let ctx = canvas.getContext('2d');
    let img = new Image();

    img.onload = function () {
        canvas.setAttribute('width', destWidth);
        canvas.setAttribute('height', destHeight);
        ctx.drawImage(img, offsetX, offsetY, resultWidth, resultHeight, 0, 0, resultWidth, resultHeight);
        callback(canvas.toDataURL());
    }
    img.src = base64Image;
}
```

**Comment by marcusdiy** [comment]
Yep, we are doing the same but doing it with js slows down the process quite a bit...
